packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts

```diff
@@ -137,8 +138,9 @@ export class EcsTask implements events.IRuleTarget {
       this.securityGroups = props.securityGroups;
       return;
     }
+
     let securityGroup = props.securityGroup || this.taskDefinition.node.tryFindChild('SecurityGroup') as ec2.ISecurityGroup;
-    securityGroup = securityGroup || new ec2.SecurityGroup(this.taskDefinition, 'SecurityGroup', { vpc: this.props.cluster.vpc });
+    securityGroup = securityGroup || new ec2.SecurityGroup(this.taskDefinition as unknown as Construct, 'SecurityGroup', { vpc: this.props.cluster.vpc });
```

Comment:
This is a little scary. 

Please replace with a 

```ts
if (!Construct.isConstruct(this.taskDefinition)) {
  throw new Error(...);
```

Given our classes this casting should work, but someone is free to implement an `ITaskDefinition` which is not a Construct. Yes, our inheritance hierarchy is a little wonky here :).

OTOH, I wonder if `this.taskDefinition.node.host` would also work...

Reply:
P.S.: The actual change of passing an imported TaskDefinition to an event target also needs a test

Reply:
1. Make sense. Will replace the casting with the check.
2. Any suggestion on verifying `this.taskDefinition.node.host`?
   1. Similarly, skinny85@ gave a callout, asking for validating if [security group](https://github.com/aws/aws-cdk/blob/98ab29206903a235331a9a0b4b38a5e6ad6d89dc/packages/%40aws-cdk/aws-events-targets/lib/ecs-task.ts#L140) can be accessed after we switch to use ITaskDefinition. Unsure how to do so tho. Any suggestions?
3. Will write a test for passing imported TaskDefinition.

Reply:
> 2\. Any suggestion on verifying `this.taskDefinition.node.host`

Try it and see if it compiles 😊. If it does I'm satisfied.

> Similarly, skinny85@ gave a callout, asking for validating if security group can be accessed after we switch to use ITaskDefinition. Unsure how to do so tho. Any suggestions?

Your test will confirm. If it doesn't throw then that is good enough. What is most likely going to happen is that new securitygroups are going to be created for this task... which is fine.

Looking at it, `EcsTask` should probably also implement `ec2.IConnectable`, otherwise it will be hard to configure the SGs. But you've already put so much effort into this change, I would be okay if you left this for someone else to pick up.